worlds/inscryption/Rules.py

```diff
@@ -0,0 +1,189 @@
+        return state.has("Inspectometer Battery", self.player)
+
+    def has_gems_and_battery(self, state: CollectionState) -> bool:
+        return state.has("Gems Module", self.player) and self.has_inspectometer_battery(state)
+
+    def has_pelts(self, count: int) -> Callable[[CollectionState], bool]:
+        return lambda state: state.has("Holo Pelt", self.player, count) and self.has_gems_and_battery(state)
+
+    def has_mycologists_boss_requirements(self, state: CollectionState) -> bool:
+        return state.has("Mycologists Holo Key", self.player) and self.has_transcendence_requirements(state)
+
+    def has_bone_lord_room_requirements(self, state: CollectionState) -> bool:
+        return state.has("Bone Lord Holo Key", self.player) and self.has_inspectometer_battery(state)
+
+    def has_battery_and_quill(self, state: CollectionState) -> bool:
+        return state.has("Quill", self.player) and self.has_inspectometer_battery(state)
+
+    def has_transcendence_requirements(self, state: CollectionState) -> bool:
+        return state.has("Quill", self.player) and self.has_gems_and_battery(state)
+
+    def has_act2_requirements(self, state: CollectionState) -> bool:
+        return state.has("Film Roll", self.player)
+
+    def has_act3_requirements(self, state: CollectionState) -> bool:
+        return self.has_act2_requirements(state) and self.has_all_epitaph_pieces(state) and \
+            self.has_camera_and_meat(state) and self.has_monocle(state)
+
+    def has_epilogue_requirements(self, state: CollectionState) -> bool:
+        return self.has_act3_requirements(state) and self.has_transcendence_requirements(state)
+
+    def set_all_rules(self) -> None:
+        multiworld = self.world.multiworld
+        if self.world.options.goal.value != Goal.option_first_act:
+            multiworld.completion_condition[self.player] = self.has_epilogue_requirements
+        else:
+            multiworld.completion_condition[self.player] = self.has_act2_requirements
+        for region in multiworld.get_regions(self.player):
+            if self.world.options.goal.value == Goal.option_full_story_in_order:
+                if region.name in self.region_rules:
+                    for entrance in region.entrances:
+                        entrance.access_rule = self.region_rules[region.name]
+            for loc in region.locations:
+                if loc.name in self.location_rules:
+                    loc.access_rule = self.location_rules[loc.name]
+
+    def set_painting_rules(self) -> None:
+        region = self.world.multiworld.get_region("Act 1", self.player)
+        loc = next((x for x in region.locations if x.name == "Act 1 - Painting 2"), None)
+        if loc is not None:
+            loc.access_rule = self.has_useful_act1_items
+        loc = next((x for x in region.locations if x.name == "Act 1 - Painting 3"), None)
+        if loc is not None:
+            loc.access_rule = self.has_useful_act1_items
```

Comment:
I'm again unsure why this can ever be `None` and isn't just:
```suggestion
    def set_painting_rules(self) -> None:
        set_rule(self.world.get_location("Act 1 - Painting 2"), self.has_useful_act1_items)
        set_rule(self.world.get_location("Act 1 - Painting 3"), self.has_useful_act1_items)
```

It seems like this is trying to prevent certain option combinations from failing, but if that's the case, you should simply be checking for those options and throwing an OptionError in `generate_early` instead of disregarding the options a player used